src/Endpoints/ScheduleMessages.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace MailerSend\Endpoints;
+
+use Assert\Assertion;
+use MailerSend\Common\Constants;
+use MailerSend\Helpers\GeneralHelpers;
+
+class ScheduleMessages extends AbstractEndpoint
+{
+    protected string $endpoint = 'message-schedules';
+
+    /**
+     * @throws \Psr\Http\Client\ClientExceptionInterface
+     * @throws \JsonException
+     * @throws \MailerSend\Exceptions\MailerSendAssertException
+     */
+    public function getAll(?string $domainId = null, ?string $status = null, ?int $page = null, ?int $limit = Constants::DEFAULT_LIMIT): array
```

Comment:
We need to add status constants into Constants.php for statuses and add validation for this field.